db/de.go

```diff
@@ -105,7 +105,23 @@ func (d *DEDatabase) AddUserDataUsageBatch(context context.Context, usages map[s
 		placeholders = append(placeholders, "(?::text, ?::bigint)")
 		startargs = append(startargs, usr, usg)
 	}
-	startcte := "WITH new_usages (username, usage) AS (VALUES " + strings.Join(placeholders, ",") + ")"
+	nonzero_usages := "new_nonzero_usages (username, usage) AS (VALUES " + strings.Join(placeholders, ",") + ")"
+
+	// Add a 0 for any user whose most recent total is > 0 but who doesn't appear in the batch we got from the ICAT
+	new_usages2, nuargs, err := psql.Select().
+		Column("us.username").
+		Column("?", 0).
+		From(d.Table("user_data_usage", "udu")).
+		Join(fmt.Sprintf("%s ON (us.id = udu.user_id)", d.Table("users", "us"))).
+		LeftJoin("new_nonzero_usages ON users.username = new_nonzero_usages.username").
+		Where(squirrel.Gt{"total": 0}).
+		Where("us.username BETWEEN ? AND ?", start, end).
+		Where("time = (SELECT MAX(time) FROM user_data_usage u2 WHERE u2.user_id = udu.user_id)").
+		ToSql()
+	startargs = append(startargs, nuargs...)
+	new_usages := "new_usages (username, usage) AS (SELECT username, usage from new_nonzero_usages UNION ALL " + new_usages2 + ")"
+
+	startcte := "WITH " + nonzero_usages + ", " + new_usages
```

Comment:
Adding this here should that the new usage values of 0 will be returned from this function, then returned from the function in the coordination.go file, whose return value is used in the AMQP handler to send the usage update messages out.